bson/doc.go

```diff
@@ -70,8 +70,9 @@
 //     otherwise.
 //  4. int64 marshals to BSON int64.
 //  5. uint8 and uint16 marshal to a BSON int32.
-//  6. uint, uint32, and uint64 marshal to a BSON int32 if the value is between math.MinInt32 and math.MaxInt32,
-//     inclusive, and BSON int64 otherwise.
+//  6. When encoding with EncodeContext.MinSize set to true, uint, uint32, and uint64 will marshal to a BSON int32 if
+//     the value falls within the inclusive range of math.MinInt32 to math.MaxInt32. Otherwise, they will marshal to a
+//     BSON int64. See Encoder examples for usage of EncodeContext.MinSize.
```

Comment:
`EncodeContext.MinSize` is deprecated, so we shouldn't recommend that people use it. They should use `Encoder.IntMinSize` instead.